python/lsst/analysis/tools/tasks/catalogMatch.py

```diff
@@ -178,262 +112,175 @@ class CatalogMatchConfig(pipeBase.PipelineTaskConfig, pipelineConnections=Catalo
+        `loadedRefCat` : astropy.table.Table
+            The loaded reference catalog
+        `bands` : list
+            A list of bands to apply the selectors in
 
         Returns
         -------
-        `pipeBase.Struct` containing:
-            matchedCat : `pandas.core.frame.DataFrame`
-                Catalog containing the matched objects with all columns from
-                the original input catalogs, with the suffix "_ref" or
-                "_target" for duplicated column names, plus a column with the
-                angular separation in arcseconds between matches.
+        `matchedCatalog` : astropy.table.Table
+
+        Notes
+        -----
+        Performs an RA/Dec match that returns the closest match
+        within the match radius which defaults to 1.0 arcsecond.
+        Applies the suffix, _target, to the catalog being matched
+        and _ref to the reference catalog.
         """
         # Apply the selectors to the catalog
         mask = np.ones(len(catalog), dtype=bool)
-        for selector in self.config.selectorActions:
-            mask &= selector(catalog, bands=self.config.bands)
-
         for selector in self.config.sourceSelectorActions:
-            mask &= selector(catalog, band=self.config.selectorBand).astype(bool)
+            for band in self.config.selectorBands:
+                mask &= selector(catalog, band=band).astype(bool)
 
         targetCatalog = catalog[mask]
-        targetCatalog = targetCatalog.reset_index()
 
-        if (len(targetCatalog) == 0) or (len(self.refCat) == 0):
-            matches = pipeBase.Struct(
-                refMatchIndices=np.array([]), targetMatchIndices=np.array([]), separations=np.array([])
-            )
+        if (len(targetCatalog) == 0) or (len(loadedRefCat) == 0):
+            refMatchIndices = np.array([], dtype=np.int64)
+            targetMatchIndices = np.array([], dtype=np.int64)
+            dists = np.array([], dtype=np.float64)
         else:
-            # Run the matcher
-            matches = self.matcher.run(self.refCat, targetCatalog)
-
-        # Join the catalogs for the matched catalogs
-        refMatches = self.refCat.iloc[matches.refMatchIndices].reset_index()
-        sourceMatches = targetCatalog.iloc[matches.targetMatchIndices].reset_index()
-        matchedCat = sourceMatches.join(refMatches, lsuffix="_target", rsuffix="_ref")
-
-        separations = pd.Series(matches.separations).rename("separation")
-        matchedCat = matchedCat.join(separations)
-
-        return pipeBase.Struct(matchedCatalog=matchedCat)
-
-    def setRefCat(self, skymap, tract):
-        """Make a reference catalog with coordinates in degrees
-
-        Parameters
-        ----------
-        skymap : `lsst.skymap`
-            The skymap used to define the patch boundaries.
-        tract : int
-            The tract corresponding to the catalog data.
+            # Run the matcher.
+
+            # This all assumes that everything is in degrees.
```

Comment:
Source catalogs are in radians! Did I miss these being converted somewhere? The reference catalogs are too natively, but I can't remember what the reference object loader returns.